devel/cctools/Portfile

```diff
@@ -134,11 +139,6 @@ if { ![some_llvm_variant_set] && ![variant_isset xcode] && ![variant_isset xtool
     }
 }
 
-if {[variant_isset llvm34] && ${os.major} < 11} {
-    # This port is used by clang-3.4 to bootstrap libcxx
-    configure.cxx_stdlib    libstdc++
-}
```

Comment:
This reintroduces a circular dependency, and even if it didn't, has nothing to do with switching to github master_sites.

Reply:
@jmroot nope, it doesn't: https://github.com/macports/macports-ports/blob/master/lang/libcxx/Portfile#L88-L96

Reply:
Yes, it does.
```
% port deps cctools configure.cxx_stdlib=libc++
Full Name: cctools @949.0.1_2+llvm34
Build Dependencies:   libunwind-headers, clang-11
Library Dependencies: llvm-3.4, libcxx
% port deps clang-11                           
Full Name: clang-11 @11.1.0_7+defaultlibcxx+emulated_tls
Extract Dependencies: xz
Build Dependencies:   cmake, cctools, python310, clang-9.0, legacy-support
Library Dependencies: libxml2, libomp, llvm-11, libcxx
Runtime Dependencies: clang_select, ld64, cctools
```

Reply:
@jmroot nope, it doesn't. It uses `clang-11-bootstrap`, not `clang-11`.

Reply:
When I'm running `port rdeps cctools` on clean 10.6 on local root `a02098b007bab105860b9faf7a897d24deafaa4f` I do have:
```
The following ports are dependencies of cctools @949.0.1_3+llvm34:
  libunwind-headers
    xz-bootstrap
  clang-11
    xz
      gettext
        clang-3.4
          cctools
            llvm-3.4
              python27-bootstrap
              libffi
                expect
                  automake
                    autoconf
                      m4
                        clang-3.7
                          llvm-3.7
                            libedit
                              ncurses
                                libcxx
                                  clang-11-bootstrap
                                    cmake-bootstrap
                                    gcc10-bootstrap
                                    libstdcxx_clang_fix
                            zlib
                            python27
                              pkgconfig
                                libiconv
                                  gperf
                              bzip2
                              db48
                              expat
                              gettext-runtime
                              sqlite3
                              openssl11
                              python_select
                              python2_select
                            perl5
                              perl5.34
                                gdbm
                                  readline
                            llvm_select
                          clang_select
                          ld64
                            ld64-127
                              libmacho-headers
                  tcl
                dejagnu
        libtextstyle
        gettext-tools-libs
    cmake
      curl
        brotli
        libidn2
          libunistring
            libtool
        libpsl
          python310
            openssl
              openssl3
            python3_select
        zstd
          lz4
        curl-ca-bundle
          unzip
      libarchive
        libxml2
          icu
        lzo2
        libb2
      libuv
        legacy-support
    clang-9.0
      libomp
      llvm-9.0
        xar
    llvm-11
```

Reply:
As a proove and to save time I've installed from binary archive a bootstrap ports: `sudo port install --unrequested clang-11-bootstrap cmake-bootstrap gcc10-bootstrap python27-bootstrap`.

After that I've started to build `cctools` from sources:
```
snow-leopard:~ catap$ sudo port install --unrequested clang-11-bootstrap cmake-bootstrap gcc10-bootstrap python27-bootstrap
--->  Computing dependencies for clang-11-bootstrap
The following dependencies will be installed:  libstdcxx_clang_fix
Continue? [Y/n]: y
--->  Fetching archive for libstdcxx_clang_fix
--->  Attempting to fetch libstdcxx_clang_fix-0.0.1_0.any_any.noarch.tbz2 from http://packages.macports.org/libstdcxx_clang_fix
--->  Attempting to fetch libstdcxx_clang_fix-0.0.1_0.any_any.noarch.tbz2.rmd160 from http://packages.macports.org/libstdcxx_clang_fix
--->  Installing libstdcxx_clang_fix @0.0.1_0
--->  Activating libstdcxx_clang_fix @0.0.1_0
--->  Cleaning libstdcxx_clang_fix
--->  Fetching archive for clang-11-bootstrap
--->  Attempting to fetch clang-11-bootstrap-11.1.0_4+emulated_tls+universal.darwin_10.i386-x86_64.tbz2 from http://packages.macports.org/clang-11-bootstrap
--->  Attempting to fetch clang-11-bootstrap-11.1.0_4+emulated_tls+universal.darwin_10.i386-x86_64.tbz2.rmd160 from http://packages.macports.org/clang-11-bootstrap
--->  Installing clang-11-bootstrap @11.1.0_4+emulated_tls+universal
--->  Activating clang-11-bootstrap @11.1.0_4+emulated_tls+universal
--->  Cleaning clang-11-bootstrap
--->  Fetching archive for cmake-bootstrap
--->  Attempting to fetch cmake-bootstrap-3.9.6_0.darwin_10.x86_64.tbz2 from http://packages.macports.org/cmake-bootstrap
--->  Attempting to fetch cmake-bootstrap-3.9.6_0.darwin_10.x86_64.tbz2.rmd160 from http://packages.macports.org/cmake-bootstrap
--->  Installing cmake-bootstrap @3.9.6_0
--->  Activating cmake-bootstrap @3.9.6_0
--->  Cleaning cmake-bootstrap
--->  Fetching archive for gcc10-bootstrap
--->  Attempting to fetch gcc10-bootstrap-10.3.0_5+universal.darwin_10.i386-x86_64.tbz2 from http://packages.macports.org/gcc10-bootstrap
--->  Attempting to fetch gcc10-bootstrap-10.3.0_5+universal.darwin_10.i386-x86_64.tbz2.rmd160 from http://packages.macports.org/gcc10-bootstrap
--->  Installing gcc10-bootstrap @10.3.0_5+universal
--->  Activating gcc10-bootstrap @10.3.0_5+universal
--->  Cleaning gcc10-bootstrap
--->  Fetching archive for python27-bootstrap
--->  Attempting to fetch python27-bootstrap-2.7.18_10.darwin_10.x86_64.tbz2 from http://packages.macports.org/python27-bootstrap
--->  Attempting to fetch python27-bootstrap-2.7.18_10.darwin_10.x86_64.tbz2.rmd160 from http://packages.macports.org/python27-bootstrap
--->  Installing python27-bootstrap @2.7.18_10
--->  Activating python27-bootstrap @2.7.18_10
--->  Cleaning python27-bootstrap
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
--->  Some of the ports you installed have notes:
  clang-11-bootstrap has the following notes:
    To use this bootstrap version of clang instead of the default compiler, add the  following lines to the Portfile:
    
    depends_build-append    port:clang-11-bootstrap
    configure.cc            ${prefix}/libexec/clang-11-bootstrap/bin/clang
    configure.cxx           ${prefix}/libexec/clang-11-bootstrap/bin/clang++
  cmake-bootstrap has the following notes:
    To use this bootstrap version of cmake instead of the usual cmake port, add the following lines to the Portfile:
    
    depends_build-replace  path:bin/cmake:cmake port:cmake-bootstrap
    configure.cmd          ${prefix}/libexec/cmake-bootstrap/bin/cmake
  gcc10-bootstrap has the following notes:
    To use this bootstrap version of gcc instead of the default compiler, add the following lines to the Portfile:
    
    depends_lib-append      port:gcc10-bootstrap
    
    configure.cc            ${prefix}/libexec/gcc10-bootstrap/bin/gcc
    configure.cxx           ${prefix}/libexec/gcc10-bootstrap/bin/g++
    
    If you would like to build universal port with this compiler, you must use per target compiler. The easy way is using muniversal PG:
    
    PortGroup               muniversal 1.0
    
    if {${universal_possible} && [variant_isset universal]} {
        configure.cc        {}
        configure.cxx       {}
    
        foreach arch ${universal_archs_supported} {
            lappend merger_configure_env(${arch})  "CC=arch -arch ${arch} ${prefix}/libexec/gcc10-bootstrap/bin/gcc"
            lappend merger_configure_env(${arch})  "CXX=arch -arch ${arch} ${prefix}/libexec/gcc10-bootstrap/bin/g++"
        }
    } else {
        configure.cc        ${prefix}/libexec/gcc10-bootstrap/bin/gcc
        configure.cxx       ${prefix}/libexec/gcc10-bootstrap/bin/g++
    }
snow-leopard:~ catap$ sudo port -s install --unrequested cctools
Password:
--->  Computing dependencies for cctools
The following dependencies will be installed: 
 autoconf
 automake
 brotli
 bzip2
 cctools
 clang-11
 clang-3.4
 clang-3.7
 clang-9.0
 clang_select
 cmake
 cmake
 curl
 curl-ca-bundle
 db48
 expat
 gdbm
 gettext
 gettext-runtime
 gettext-tools-libs
 gperf
 icu
 ld64
 ld64-127
 legacy-support
 libarchive
 libb2
 libcxx
 libedit
 libffi
 libiconv
 libidn2
 libmacho-headers
 libomp
 libpsl
 libtextstyle
 libtool
 libunistring
 libunwind-headers
 libuv
 libxml2
 llvm-11
 llvm-3.4
 llvm-3.7
 llvm-9.0
 llvm_select
 lz4
 lzo2
 m4
 ncurses
 openssl
 openssl11
 openssl3
 perl5
 perl5.34
 pkgconfig
 python27
 python2_select
 python310
 python3_select
 python_select
 readline
 sqlite3
 xar
 xz
 xz
 xz-bootstrap
 zlib
 zstd
Continue? [Y/n]: y
--->  Fetching distfiles for xz-bootstrap
--->  Attempting to fetch xz-5.4.2.tar.bz2 from http://distfiles.macports.org/xz
--->  Verifying checksums for xz-bootstrap
```

=> no loop.